### ISSUE TITLE
[vue/valid-template-root] for functional component

### ISSUE THREAD

**Issue by service-paradis** [vuejs/eslint-plugin-vue#977]
**Tell us about your environment**	
* **ESLint version:**	5
* **eslint-plugin-vue version:**	5.2.3
* **Node version:**	10.15.0

**Please show your full configuration:**

```json
module.exports = {
  root: true,
  env: {
    node: true
  },
  'extends': [
    'plugin:vue/recommended',
    '@vue/typescript'
  ],
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off'
  },
  parserOptions: {
    parser: '@typescript-eslint/parser'
  },
  overrides: [
    {
      files: [
        '**/__tests__/*.{j,t}s?(x)'
      ],
      env: {
        jest: true
      }
    }
  ]
}

```

**What did you do?**	

```vue
<template functional>
  <div>Child 1</div>
  <div>Child 2</div>
</template>
```

**What did you expect to happen?**
No lint error since multiple element is valid in a functional component

**What actually happened?**
Error:
> [vue/valid-template-root]
The template root requires exactly one element.

**Comment by xepher0coded** [comment]
@service-paradis, could you tell me what you did to solve this?
